### PR TITLE
wpf: add width/height checks when resizing the terminal

### PR DESCRIPTION
**Pull request by javierdlg**
We are getting some watson crash reports that the terminal is attempting
to resize to `(0, 0)`. This change makes it so that we prevent such
resizing and if so, throw an exception before we reach native code.

This commit adds resizing checks that prevent resizing the terminal WPF
control to a size of `(0, 0)`

### CODE REVIEW

**Review by RyanToth3 — APPROVED** [review]

**Comment by WSLUser** [comment]
Looks like you are using a fork prior to Super Linter getting added. Can you rebase/merge master to your fork?

**Comment by DHowett** [comment]
Because of the conflict, Javier will need to merge master anyway. Be aware though: the linter doesn't lint C# files, and we use stylecop for that.

**Comment by zadjii-msft** [comment]
> . Be aware though: the linter doesn't lint C# files, and we use stylecop for that.

> image

Well, that's sure to conflict, won't it.

**Review by zadjii-msft — APPROVED** [review]
Does resizing to (0,0) cause the crashes, or does resizing to 0 in either dimension (i.e. (w,0) or (0,h)) also cause a crash?

**Comment by javierdlg** [comment]
@zadjii-msft I haven't tested (0, #) but when set to (0,0) it crashes VS since an exception is raised in the native side. We got the details via a Watson dump and it shows more details there.

You can find more information and the dump [here](https://watsonportal.microsoft.com/Failure?FailureSearchText=f98f1c05-23c1-32d4-4ae5-f99e3510a7c9&MaxRows=100)

File: src/cascadia/WpfTerminalControl/TerminalContainer.cs [hunk 1]
```diff
@@ -161,6 +164,13 @@ internal string GetSelectedText()
         /// <param name="renderSize">Size of the rendering window.</param>
         internal void Resize(Size renderSize)
         {
+            if (renderSize.Width == 0 || renderSize.Height == 0)
+            {
+                throw new ArgumentException(nameof(renderSize), "Terminal column or row count cannot be 0.");
+            }
+
+            var dpiScale = VisualTreeHelper.GetDpi(this);
```
Comment (DHowett) [inline src/cascadia/WpfTerminalControl/TerminalContainer.cs:172]: i don't see why we need this line..

**Review by DHowett — CHANGES_REQUESTED** [review]
Just the one question

File: src/cascadia/WpfTerminalControl/TerminalContainer.cs (hunk 1 above)
Comment (javierdlg) [reply src/cascadia/WpfTerminalControl/TerminalContainer.cs:172]: Looks like it came from a bad merge. Pushing a fix...

**Review by DHowett — APPROVED** [review]

**Pull request merged**